### PR TITLE
Instance: let a root device of a running VM be set only if the key is `size.state`

### PR DESCRIPTION
**Pull request by gabrielmougard**
For example, having a VM called `v1` that is running, we'd like to be able to do:

* `lxc config device set v1 root size.state=<some_size>` 

If the created VM already has `migration.stateful=true` (but with a default `size.state` that is too small to do any stateful stop), this would allow a user to to change the size.state of the root disk and do a stateful stop without having to stop the VM beforehand.

The side effect of doing this is that the `root` storage device config is moved in the instance local config instead of being in the expanded config.
___ 

* [x] `lxc config device set v1 root size.state=<some_size>` on a running VM works
* [x] Experimenting with `lxc stop v1 --stateful`
* [x] Experimenting with `lxc copy v1 remote2:v2`
* [x] Experiment with `lxc move v1 --target host2`

### CODE REVIEW

File: lxc/config_device.go [hunk 1]
```diff
@@ -620,12 +621,49 @@ func (c *cmdConfigDeviceSet) Run(cmd *cobra.Command, args []string) error {
 
 		dev, ok := inst.Devices[devname]
 		if !ok {
-			_, ok = inst.ExpandedDevices[devname]
+			expandedDev, ok := inst.ExpandedDevices[devname]
 			if !ok {
 				return fmt.Errorf(i18n.G("Device doesn't exist"))
 			}
 
-			return fmt.Errorf(i18n.G("Device from profile(s) cannot be modified for individual instance. Override device or modify profile instead"))
```
Comment (tomponline) [inline lxc/config_device.go:628]: We shouldnt be removing this check.
Instead if the root disk is from the profile we'd expect the users to use `lxc config device override <instance> root size.state=`

Also we shouldnt only be doing validation on the client.

**Comment by gabrielmougard** [comment]
@tomponline Then, with https://github.com/canonical/lxd/pull/12912, I can safely do the following:

```bash
# This works now
lxc launch ubuntu:jammy v1 --vm --config migration.stateful=true
# Then overriding the root disk while the VM is running already works with
lxc config device override v1 root size.state=2GiB
# Then stateful operations work fine
lxc stop v1 --stateful
lxc move v1 --target host2
```

What is the use of this PR now?

**Comment by tomponline** [comment]
> What is the use of this PR now?

possibly nothing - you tell me :)

**Comment by gabrielmougard** [comment]
If we already can override the device on a live VM, I'd say this PR is useless. So I'll close it.

**Pull request closed without merging**